evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py

```diff
@@ -475,54 +592,24 @@ def _add_custom_index(answer, index_best, index_worst, output_format):
                                                        num_to_explain=1, output_format=output_format)
     if output_format == "text":
         compare_two_tables(best_worst_report.splitlines(), answer.splitlines())
+    elif output_format == "dataframe":
+        # Check dataframes equal without caring about column order
+        assert sorted(best_worst_report.columns.tolist()) == sorted(answer.columns.tolist())
+        pd.testing.assert_frame_equal(best_worst_report, answer[best_worst_report.columns])
```

Comment:
Won't assert_frame_equal compare the columns as well?  L597 might be redundant?

Reply:
Yes but in the event `best_worst_report` has fewer columns than expected, the check will pass because `answer[best_worst_report.columns]` (needed to reorder the columns) will subset to the columns in `best_worst_report`. 

The root cause is that I modify the `index_id` column at the beginning of the test to add the expected custom index and this unfortunately messes up the column order. I don't think there's a way to get assert_frame_equal to ignore the column order so this two-step check verifies the same columns are present in both and that they contain the same values. I think this is ok given how many different scenarios we're testing. The alternative would be to manually specify the answer for each test but that would be hairy.